clang/test/CheckedC/inferred-bounds/bounds-context.c

```diff
@@ -300,16 +300,14 @@ void assign1(array_ptr<int> arr : count(1)) { // expected-note {{(expanded) decl
 // Assignment to a variable used in other variables' bounds
 void assign2(
   array_ptr<int> a : count(len - 1), // expected-note {{(expanded) declared bounds are 'bounds(a, a + len - 1)'}}
-  char b nt_checked[0] : count(len), // expected-note {{(expanded) declared bounds are 'bounds(b, b + len)'}}
+  char b nt_checked[0] : count(len),
```

Comment:
This change on line 303 and lines 311 and 312 do not seem to be caused by the implementation of invertibility. It will be great if you could give some context for this change.

Reply:
These changes are a direct result of invertibility. `b` is a null-terminated array which uses `len` in its bounds and `len` is modified. Without invertibility we would have killed the bounds of `b` and issued the warning/note messages. But with invertibility we can "adjust" the bounds of `b` to `bounds(b, b + len - 3)`.

Reply:
Filed https://github.com/microsoft/checkedc-clang/issues/1138 to track the invalid declaration of `nt_checked` array of size 0.